persistence/src/main/scala/com/mesosphere/usi/repository/ReservationRecordRepository.scala

```diff
@@ -5,7 +5,4 @@ import com.mesosphere.usi.core.models.{ReservationId, ReservationRecord}
 /**
   * Repository for [[ReservationRecord]].
   */
-trait ReservationRecordRepository extends RecordRepository {
-  override type Record = ReservationRecord
-  override type RecordId = ReservationId
-}
+trait ReservationRecordRepository extends RecordRepository[ReservationRecord, ReservationId]
```

Comment:
@meln1k, what was the reason to use `type` instead of a template?

Reply:
I don't want to carry generic parameters everywhere since they look messy and don't provide real value in this case. Once they're introduced, all code that works with the repository will have to deal with them, and it's going to complicate things without providing any real benefits.

Reply:
Not sure. The type alias can cause confusion. Especially since it's could be used in the implementation and one would have two `Record` and `RecordId` types.

Reply:
I don't think there would be code that uses the `Repository` trait directly. I might be wrong though. Of course `repo: Repository[_, _]` would work. It actually gives more control in case we want `repo: Repository[CoreId, _]`